Read metadata from PJN file into hivproj data

r-ash opened the pull request:
* hivproj data structure needs to included some metadata read from .PJN file, this pull request adds functions to extract those bits of metadata and a test

richfitz commented on R/file_utils.R:
```diff
@@ -50,13 +50,14 @@ readlines_from_path <- function(pjnz_path, extension) {
 #'
-#' @param pjnz_path Path to zip containing DP file.
+#' @param pjnz_path Path to PJNZ zip file.
+#' @param extension The file extension to look for.
```
Tiny comment, but I would recommend giving an example here to remove ambiguity about whether `.` is required in the extension (`.DP` vs `DP`)

r-ash commented:
> Sorry for the delay, took me a while to realise that this generalised the existing functionality rather than removing it.
> 
> If you wanted to support `get_dp_data` I think it would be totally reasonable to export that function (any others) with an implementation simply as
> 
> ```r
> get_dp_data <- function(pjnz_path) {
>   get_pjnz_csv_data(pjnz_path, "DP")
> }
> ```
> 
> which would leave previous tests alone, and any previous uses. Matter of taste only of course, and it depends on the number of different formats used. One advantage of that is it cuts down on the number of errors in uses because someone has used `dp`, `.DP`, `Dp`, etc

I like this, I feel its probably easier to read `get_dp_data` and `get_pjn_data` too. I've updated to have separate functions for those. 

I don't expect that `get_pjnz_csv_data` or indeed `get_dp_data`, `get_pjn_data` should be exported. I'm not expecting anyone to need to call them outside of the package. I've added the `@keywords internal` tag which I'd missed off previously.
